Feature/login page ui 20191115

riverandeye opened the pull request:
nanum-square 폰트를 적용하였습니다.
LogoSVG의 path가 svg 컴포넌트에 적절히 배치되도록 viewBox를 수정하였습니다. 
데스크탑, 모바일 환경을 모두 고려한 로그인 페이지 UI를 작성하였습니다.

https://client.riverandeye.now.sh/
https://client.riverandeye.now.sh/login
결과는 여기에서 확인하실 수 있습니다.

riverandeye committed:
[client] | fix(LogoSVG): svg props 수정 및 viewBox prop 추가

SVG의 Path가 SVG 컴포넌트의 중심에 배치되게 하기 위해 viewBox prop을 추가 작성하였습니다.
SVG의 width와 height를 조정할 수 있게끔 설정하였습니다.

riverandeye committed:
[client] feat(GlobalStyles): nanum-square 폰트 추가

_document.tsx에 폰트 cdn을 추가하였고, GlobalStyles 컴포넌트에서 폰트를 지정하였습니다.

riverandeye committed:
[client] refactor(components): nav.js 삭제

next.js 초기 설정시 생성된 파일을 삭제하였습니다.

riverandeye committed:
[client] feat(svgs) : 깃헙 옥토캣 SVG 추가

회원가입 및 로그인 페이지에 필요한 옥토캣 SVG를 추가하였습니다.

riverandeye committed:
[client] feat(views) : 로그인 페이지 추가

상단 AppBar를 누를 시 이동되는 로그인 페이지를 추가하였습니다.

riverandeye committed:
[client] fix(GlobalStyles): nanum-square 폰트 제거

폰트를 추가하는 방식이 부적절하여 해당 커밋을 제거합니다.
This reverts commit 0a12c905f1ddc2d2f8247e5cbc8ff3b038a7342b.

hyokualexkwon commented on packages/client/views/Login/index.tsx:
```diff
@@ -0,0 +1,29 @@
+
+const Login: React.FunctionComponent = () => {
+  return (
+    <S.Container>
```
가장 바깥의 스타일드 컴포넌트는 Login이라고 명시하는게 좋을것 같습니다. 컴포넌트 최상단 스타일드 컴포넌트 네이밍은 이런 규칙을 따르도록 정했었습죠~

 S.Login이라고 명명하면 될것 같은데요?

riverandeye replied:
주의하도록 하겠습니다.

hyokualexkwon commented on packages/client/views/Login/styles.tsx:
```diff
@@ -0,0 +1,52 @@
+`;
+
+export const Message = styled.div`
+  width: 31.6rem;
```
316px은 왜 주신건가요? 해당 텍스트는 화면 넓이에 따라서 유동적으로 변합니다. 모바일에서는 넓이 제한이 없습니다.

alexKwonIsAwesome commented on packages/client/views/Login/styles.tsx:
```diff
@@ -0,0 +1,52 @@
+  margin-top: 3.1rem;
+  margin-bottom: 20.8rem;
+  @media only screen and (min-width: ${BREAKPOINT}px) {
+    width: 43rem;
```
그렇죠 모바일과 달리 데스크톱의 경우에는 최대 넓이가 있습니다. 좋네요!

Atercatus replied:
로그인 페이지에는 DESKTOP 기준 버튼외에는 넓이가 고정된게 없다고 하시네요~

hyokualexkwon commented on packages/client/views/Login/styles.tsx:
```diff
@@ -0,0 +1,52 @@
+  text-align: center;
+  font-size: 1.5rem;
+  color: white;
+  opacity: 75%;
```
디자인 스펙에 따르면 투명도는 없었습니다.

hyokualexkwon commented on packages/client/views/Login/styles.tsx:
```diff
@@ -0,0 +1,52 @@
+  width: 31.6rem;
+  height: 10rem;
+  text-align: center;
+  font-size: 1.5rem;
```
디자인 스펙에 따르면 폰트는 16px입니다.

hyokualexkwon commented on packages/client/views/Login/styles.tsx:
```diff
@@ -0,0 +1,52 @@
+
+export const Message = styled.div`
+  width: 31.6rem;
+  height: 10rem;
```
높이는 컨텐츠의 길이에 따라 유동적으로 변합니다. 고정값을 주면 안됩니다.

hyokualexkwon commented on packages/client/views/Login/styles.tsx:
```diff
@@ -0,0 +1,52 @@
+  flex-direction: row;
+  align-items: center;
+  justify-content: center;
+  width: 32rem;
```
모바일 화면에서는 넓이가 유동적으로 변합니다. 100%가 되어야 해요. 데스크톱에서만 max-width로 넓이 제한을 주시면 됩니다.

Atercatus replied:
데스크톱 화면에서만 최대 넓이를 고정하고, 모바일에서는 100%에서 화면의 좌우 여백만 20을 주어야 할 거 같습니다.

hyokualexkwon commented on packages/client/views/Login/styles.tsx:
```diff
@@ -0,0 +1,52 @@
+  height: 5.4rem;
+  background-color: #222222;
+  text-align: center;
+  border-radius: 3.4rem;
```
완벽히 둥근 사각형에는 border-radius 100%를 주시면 됩니다.

riverandeye replied:
border-radius를 100%를 주니 아예 타원이 되어 버렸어요..

hyokualexkwon replied:
ㅈㅅ합니다 그냥 가주세용! ㅋㅋ

riverandeye committed:
[client] | fix(Login): 로그인 페이지 CSS 및 컴포넌트 수정사항 반영

PR 리뷰에 제공하신 피드백 반영하였습니다.
- Login 컴포넌트를 최상위에 배치하였습니다.
- 메세지 컴포넌트 width를 100%로 변경하였습니다.
- AuthenticateButton을 그 의미에 맞게 Link로 변경하였습니다.
- 모바일 환경에서 AuthenticationLink의 width 제한을 해제하였고, 데스크탑에 반영하였습니다.
- export하는 LogoSVG 컴포넌트의 이름 변경 및 prop 으로 크기 전달하였습니다.

hyokualexkwon reviewed (approved):
수고 많으셨습니다!

KimJye reviewed (approved):
확인했습니다
